pkg/services/ssosettings/ssosettingsimpl/service.go

```diff
@@ -250,6 +254,38 @@ func (s *SSOSettingsService) encryptSecrets(ctx context.Context, settings map[st
 	return result, nil
 }
 
+// initReload starts a background process for reloading the SSO settings for all providers at a fixed interval
+// it is useful for high availability setups running multiple Grafana instances
+func (s *SSOSettingsService) initReload() {
+	go func() {
+		reloadTicker := time.NewTicker(1 * time.Minute)
+		for {
+			<-reloadTicker.C
+			s.doReload()
+		}
+	}()
+}
+
+func (s *SSOSettingsService) doReload() {
+	for provider, connector := range s.reloadables {
+		s.log.Debug("reloading SSO Settings", "provider", provider)
+
+		ctx := context.Background()
+
+		settings, err := s.GetForProvider(ctx, provider)
```

Comment:
This will do a _sql_ query for each _reloadable_, can we batch this? Alternative fetch everything stored in the db upfront and then trigger reload for each provider?

Reply:
I refactored this function to call the list method in order to get all sso settings with one sql query for all providers.

@colin-stuart since you are working on the list endpoint, can you please make sure that the list API endpoint is calling a separate service method that returns the sso settings with secrets redacted? Similar to what you did for the GET endpoint (`GetForProvider` and `GetForProviderWithRedactedSecrets`).